src/main/java/org/datavyu/plugins/ffmpeg/MpvMediaPlayer.java

```diff
@@ -0,0 +1,311 @@
+        }
+    }
+
+    @Override
+    protected void playerStop() throws MediaException {
+        int rc = mpvStop(getNativeMediaRef());
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+    }
+
+    @Override
+    protected void playerPause() throws MediaException {
+        int rc = mpvPause(getNativeMediaRef());
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+    }
+
+    @Override
+    protected void playerStepForward() throws MediaException {
+        int rc = mpvStepForward(getNativeMediaRef());
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+    }
+
+    @Override
+    protected void playerStepBackward() throws MediaException {
+        int rc = mpvStepBackward(getNativeMediaRef());
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+    }
+
+    @Override
+    protected void playerFinish() throws MediaException {
+        int rc = mpvFinish(getNativeMediaRef());
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+    }
+
+    @Override
+    protected float playerGetRate() throws MediaException {
+        float[] rate = new float[1];
+        int rc = mpvGetRate(getNativeMediaRef(), rate);
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+        return rate[0];
+    }
+
+    @Override
+    protected void playerSetRate(float rate) throws MediaException {
+        int rc = mpvSetRate(getNativeMediaRef(), rate);
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+    }
+
+    @Override
+    protected double playerGetPresentationTime() throws MediaException {
+        double[] presentationTime = new double[1];
+        int rc = mpvGetPresentationTime(getNativeMediaRef(), presentationTime);
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+        return presentationTime[0];
+    }
+
+    @Override
+    protected double playerGetFps() throws MediaException {
+        double[] framePerSecond = new double[1];
+        int rc = mpvGetFps(getNativeMediaRef(), framePerSecond);
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+        return framePerSecond[0];
+    }
+
+
+    @Override
+    protected float playerGetVolume() throws MediaException {
+        synchronized(this) {
+            if (muteEnabled)
+                return mutedVolume;
+        }
+        float[] volume = new float[1];
+        int rc = mpvGetVolume(getNativeMediaRef(), volume);
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+        // MPV Volume range is from 0 to 100 and in order to much
+        // the java native plugin range we divide by 100
+        return volume[0] / 100;
+    }
+
+    @Override
+    protected synchronized void playerSetVolume(float volume) throws MediaException {
```

Comment:
Why is this method synchronized and the others are not?

Reply:
Fixed